tool/site/client-react/src/components/sorter.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {ReactNode} from "react"
+import {ChevronLeft, ChevronRight} from "nhsuk-react-components/lib/components/icons"
+import styled from "styled-components"
+
+export interface SortConfig<T> {
+  key: keyof T
+  ascending: boolean
+}
+
+export interface Sorter<T> {
+  sortedItems: Array<T>
+  sortBy: (key: keyof T) => void
+  getIcon: (key: keyof T) => ReactNode
+}
+
+export const useSorter = <T extends unknown>(items: Array<T>, config: SortConfig<T> = null): Sorter<T> => {
+  const [sortConfig, setSortConfig] = React.useState<SortConfig<T>>(config)
+
+  const sortedItems = React.useMemo(() => {
+    const itemsCopy = [...items]
+    if (sortConfig !== null) {
+      itemsCopy.sort((a, b) => {
+        const sortValueA = a[sortConfig.key]
+        const sortValueB = b[sortConfig.key]
+        const directionValue = sortConfig.ascending ? 1 : -1
+        if (sortValueA > sortValueB) {
```

Comment:
This naive comparison seems to work alright for strings, numbers and dates. Could implement custom sorting functions but don't see the need yet